packages/aws-cdk-lib/aws-logs/lib/log-group.ts

```diff
@@ -534,6 +534,25 @@ export class LogGroup extends LogGroupBase {
       arnFormat: ArnFormat.COLON_RESOURCE_NAME,
     });
     this.logGroupName = this.getResourceNameAttribute(resource.ref);
+
+    if (props.encryptionKey) {
+      props.encryptionKey.addToResourcePolicy(new iam.PolicyStatement({
+        principals: [new iam.ServicePrincipal(`logs.${this.env.region}.amazonaws.com`)],
+        actions: [
+          'kms:Encrypt*',
+          'kms:Decrypt*',
+          'kms:ReEncrypt*',
+          'kms:GenerateDataKey*',
+          'kms:Describe*',
+        ],
+        resources: ['*'],
+        conditions: {
+          ArnLike: {
+            'kms:EncryptionContext:aws:logs:arn': `arn:${this.stack.partition}:logs:${this.env.region}:${this.env.account}:*`,
```

Comment:
should this be `log-group:<log-group-name>` instead of `*` ?

Reply:
I tried to use `ArnEquals` with the log group ARN but it created a circular dependency between the key and the log group.
Not sure if there's a workaround, I agree that this may be too permissive.

Reply:
If there is no easy workaround, I am OK with this, as the second example in the docs has `*`.

Reply:
I'm not sure if this will work @lpizzinidev but have you tried using `log-group:${this.physicalName}`? Instead of using `logGroupArn` or `logGroupName`, which are attributes of the log group.

Reply:
@kaizencc 
I tried `log-group:${this.physicalName}` and it works.
It should be safe to go with it, let me know if you think otherwise.

Reply:
It works as in you integ tested it and confirmed that the permission boundary works as expected (not just that it deploys successfully)? 

If so, I think that's the best path forward, and once thats in I'm happy to approve.

Reply:
@kaizencc 
I got it wrong.
```
conditions: {
  ArnEquals: {
    'kms:EncryptionContext:aws:logs:arn': `arn:${this.stack.partition}:logs:${this.env.region}:${this.env.account}:log-group:${this.physicalName}`,
  },
},
```
Produces this incorrect policy condition:
```
Condition: {
  ArnEquals: {
    'kms:EncryptionContext:aws:logs:arn': {
      'Fn::Join': [
        '',
        [
          'arn:',
          {
            Ref: 'AWS::Partition',
          },
          ':logs:',
          {
            Ref: 'AWS::Region',
          },
          ':',
          {
            Ref: 'AWS::AccountId',
          },
          ':log-group:',
        ],
      ],
    },
  },
},
```
It deploys successfully but the log group name is left blank.

Reply:
If you use `Lazy` does that fix it ?

Reply:
@lpizzinidev ^. Fingers crossed it does :)